app/views/bastion/layouts/assets.html.erb

```diff
@@ -1,3 +1,7 @@
+<% content_for(:head) do %>
+  <base href="<%= Bastion.config['relativeUrlRoot'] %>" />
```

Comment:
The lack of base href caused the infinite reload behavior. Addressing this revealed other issues around templates and route generation.